src/openttd.cpp

```diff
@@ -1068,6 +1068,14 @@ void SwitchToMode(SwitchMode new_mode)
 			break;
 		}
 
+		case SM_JOIN_GAME:
+			/* Used in case a (re)connect is triggered from inside a (network) game
+			 * to prevent invalid pointer dereferences. Loads the main menu and then
```

Comment:
you keep mentioning "invalid pointer dereferences" in comments, but it is not really clear why/how. Is there any chance we can be more clear in what the issue is? Maybe borrow some text from the commit message, as that does seem to explain the "why" in more detail?

Reply:
If added more information to NetworkClientJoinGame and added a reference in the documentation to there.